src/rebar3_hex.erl

```diff
@@ -29,6 +29,18 @@ task_args(State) ->
     {[{task, Task} | Args], _} = rebar_state:command_parsed_args(State),
     {Task, Args}.
 
+args(State) ->
+    {Args, _} = rebar_state:command_parsed_args(State),
+    lists:foldl(fun fold_arg/2, #{}, Args).
+
+fold_arg({K, V}, Acc) ->
+    case maps:get(K, Acc, undefined) of
+        undefined ->
+            maps:put(K, V, Acc);
+        [_|_] = Vals ->
+            maps:put(K, [V | Vals], Acc)
+    end.
```

Comment:
this fails if you get duplicate keys and the first key inserted wasn't a list since it won't match either clause. I'm unsure if the values are always meant to be list, crash as implemented here, or meant to be turned to lists when there are more than one

equivalent to: `maps:update_with(K, fun(Vals = [_|_]) -> [V|Vals] end, V, Acc)` right now; `maps:update_with(K, fun(Vals) -> [V|Vals] end, [V], Acc)` for a format that's always lists. For the switch to turning values to lists, it's fundamentally a bit risky since it's not obvious what to do for strings.

Reply:
Eh yeah, good point.

Reply:
Refactored args/1 to gather_opts/2 to be on the safe side of the tracks.